skopt/learning/gaussian_process/kernels.py

```diff
@@ -120,7 +130,11 @@ def gradient_x(self, x, X_train):
             sqrt_3_dist = sqrt(3) * dist
             f = np.expand_dims(1 + sqrt_3_dist, axis=1)
 
-            dist_expand = np.expand_dims(sqrt(3) / dist, axis=1)
+            # Gradients when dist equal to zero is zero.
```

Comment:
OK I am confused now. Some fresh eyes might be helpful.

https://github.com/scikit-optimize/scikit-optimize/blob/master/skopt/learning/gaussian_process/kernels.py#L117

The value of `f` when `X = Y` equals 1.

From here (https://github.com/scikit-optimize/scikit-optimize/blob/master/skopt/learning/gaussian_process/kernels.py#L132)

this reduces to `f_grad * g * ( 1- f )`, so it should be zero right when X equals Y?